howtos/otel/how_to_integrate_opentelemetry_with_pubsubplus_api.py

```diff
@@ -0,0 +1,150 @@
+            self._baggage = baggage.get_all(extracted_ctx)
+            
+
+            # The returned token lets you restore the previous context.
+            token = context.attach(extracted_ctx)
+            try:
+                # create a new span for every message
+                with tracer.start_as_current_span("{topic_name}_process".format(topic_name=message.get_destination_name()), kind=SpanKind.CONSUMER) as span:
+                    # set attributes for the span
+                    span.set_attribute(SpanAttributes.MESSAGING_SYSTEM, "PubSub+")
+                    span.set_attribute(SpanAttributes.MESSAGING_DESTINATION_KIND, MessagingDestinationKindValues.QUEUE.value)
+                    span.set_attribute(SpanAttributes.MESSAGING_DESTINATION, message.get_destination_name())
+                    span.set_attribute(SpanAttributes.MESSAGING_OPERATION, "process")
+                    try:
+                        print(f"Received message {message}")
+                        function_that_might_raise_exception(False)
+                        span.set_status(Status(StatusCode.OK))
+                    except Exception as ex:
+                        span.set_status(Status(StatusCode.ERROR))
+                        span.record_exception(ex)
+            finally:
+                # restore the context
+                context.detach(token)
+
+        except Exception as error:
+            self._exception_list.append(error)
+        finally:
+            if self._receiver \
+                and isinstance(self._receiver, PersistentMessageReceiver) \
+                and self._ack:
+                self._receiver.ack(message)
+            # The _receive_count metric is used to signal watching waiters that they can proceed,
+            # so we need to increment this count only after all operations have completed to avoid
+            # race conditions in testing.
+            self._receive_count += 1
+
+    @property
+    def exception_list(self) -> List[Exception]:
+        return self._exception_list
+
+    @property
+    def exception_count(self) -> int:
+        return len(self._exception_list)
+
+    @property
+    def total_message_received_count(self) -> int:
+        return self._receive_count
+
+    @property
+    def baggage(self) -> dict:
+        return self._baggage
+
+def how_to_publish_with_otel(messaging_service: MessagingService, topic_name: str):
+    """
+    This howto assumes:
+        * that the messaging service has already been created and connected
+        * that a durable exclusive queue has been created and configured to process messages with the given topic
+        * that a telemetry profile has been configured on the broker
+        * that a filter with a subscription to the given topic has been created for that telemetry profile
+        * that these resources are independently cleaned up after this function executes
```

Comment:
doesnt this howto also assumes a collector is running, whether its the solace cloud hosted one or a local collector?